src/main/java/com/google/sps/data/Guide.java

```diff
@@ -10,19 +13,45 @@
 
     public Guide(String title, String description, String content, String category){
         this.id = -1;
-        this.title = title;
-        this.description = description;
-        this.content = content;
-        this.category = category;
+
+        if(!validIndexedString(title))
```

Comment:
I would separate the validation from the object that holds this data. You want to apply the validation when a create or update request is made, but not necessarily when reading a Guide from Datastore. If you imagine that validation logic changes over time, and stored data doesn't necessarily pass all of the checks, it could be unexpected behavior to throw an exception when trying to read a Guide.

I suggest that in the servlet, you clean the strings and construct a Guide instance. Then, pass that instance to a validate method, either in a new GuideValidator class or a static method right here.

Reply:
Right, I'm going to take the static method path.